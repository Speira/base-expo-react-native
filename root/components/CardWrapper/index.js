import PropTypes from 'prop-types'
import styled from 'styled-components/native'
import BaseCardWrapper, { CardWrapperTitle, CardWrapperContent } from './style'

/**
 * CardWrapper
 * @desc ::: A wrapper for card
 *
 */
export default function CardWrapper(props) {
  const { children, onScroll, title, ...restProps } = props

  return (
    <BaseCardWrapper {...restProps}>
      <CardWrapperTitle>{title}</CardWrapperTitle>
      <CardWrapperContent onScroll={onScroll}>{children}</CardWrapperContent>
    </BaseCardWrapper>
  )
}

CardWrapper.defaultProps = {
  children: undefined,
  title: '',
  onScroll: () => null,
}
CardWrapper.propTypes = {
  children: PropTypes.node,
  title: PropTypes.string,
  onScroll: PropTypes.func,
}
