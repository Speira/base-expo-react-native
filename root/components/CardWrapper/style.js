import styled from 'styled-components/native'
import { Text, View, ScrollView } from 'react-native'

/**
 * CardWrapperTitle
 * @component
 *
 */
export const CardWrapperTitle = styled(Text)`
  flex: 0.1;
  font-size: 24px;
  font-weight: 500;
  color: ${({ theme }) => theme.COLORS.STATIC.DARK};
  text-align: center;
`
/**
 * CardWrapperContent
 * @component
 *
 */
export const CardWrapperContent = styled(ScrollView).attrs(() => ({
  contentContainerStyle: {
    flexDirection: 'row',
    flexWrap: 'wrap',
    justifyContent: 'center',
    alignItems: 'flex-start',
    alignContent: 'flex-start',
  },
}))`
  flex: 0.9;
  margin: 0 10px 10px;
  border-radius: 8px;
  background-color: ${({ theme }) => theme.COLORS.STATIC.TERTIARY};
  padding: 4px;
`
/**
 * BaseCardWrapper
 * @component
 *
 */
const BaseCardWrapper = styled(View)`
  flex: 1;
  background-color: ${({ theme }) => theme.COLORS.STATIC.SECONDARY};
  padding-top: 30px;
`

export default BaseCardWrapper
