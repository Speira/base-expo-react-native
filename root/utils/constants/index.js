// utils/constants
// description: We try to set all constants into a big one to reduce Javascript
//              constants performance issue.

export default {
  API: {
    FETCH_DATA_URL:
      'https://static.leboncoin.fr/img/shared/technical-test.json',
    PROXY: {
      NO_CORS_SERVER: 'https://cors-anywhere.herokuapp.com/',
    },
    TIMEOUT: 20000,
  },
  LABELS: {
    CLEAR_DATABASE: 'Clear the database',
    START_DATABASE: 'Create the database',
    IMVALID_DATA_FORMAT: ' Invalid data format',
    LIST: 'Item List',
    NO_CARD: 'No card to persist',
    NO_DATABASE: 'The database is not set',
    NO_RESPONSE_DATA: 'There is no data in the query response',
    REFETCH_DATA: 'Refetch Data',
    SEARCH_DATA_FAILED:
      'No data for SearchDataAPI: Please unlock the proxy below and then refetch data',
    WELCOME: 'Welcome to the ListItem app',
  },
  STATES: {
    ERROR: 'error',
    LOADING: 'loading',
  },
  STATUS: {
    DANGER: 'DANGER',
    INFO: 'INFO',
    SUCCESS: 'SUCCESS',
    WARNING: 'WARNING',
  },
  STORAGE_ITEMS: {
    DB: 'storage_list_item_app',
    CARD: 'card_items.db',
    THEME: 'app_theme_storage',
  },
  THEMES: {
    DEFAULT: {
      COLORS: {
        STATIC: {
          PRIMARY: '#EBD8C3',
          SECONDARY: '#F7E9D7',
          TERTIARY: '#FFF6EA',
          QUATERNARY: '#FFEEEE',
          LIGHT: '#f8f8fa',
          DARK: '#4d3e30',
        },
        DYNAMIC: {
          INFO: '#035397',
          SUCCESS: '#1C7947',
          WARNING: '#C36A2D',
          DANGER: '#A9333A',
        },
      },
    },
  },
}
