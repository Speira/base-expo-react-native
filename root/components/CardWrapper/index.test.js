import React from 'react'
import { shallow } from 'enzyme'
import constants from '~utils/constants'
import CardWrapper from './index'
import BaseCardWrapper, { CardWrapperTitle, CardWrapperContent } from './style'

const { THEMES } = constants

describe('render CardWrapper', () => {
  const onScroll = jest.fn()
  const wrapper = shallow(
    <CardWrapper title="test" onScroll={onScroll} theme={THEMES.DEFAUL}>
      test
    </CardWrapper>,
  )
  it('CardWrapper should be defined', () => {
    expect(wrapper).toBeDefined()
  })
  it('CardWrapper should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
  it('CardWrapper should have CardWrapperContent', () => {
    const content = wrapper.find(CardWrapperContent)
    expect(onScroll).not.toHaveBeenCalled()
    content.simulate('scroll')
    expect(onScroll).toHaveBeenCalled()
  })
})
