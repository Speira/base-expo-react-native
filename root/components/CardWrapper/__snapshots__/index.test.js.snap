// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render CardWrapper CardWrapper should match snapshot 1`] = `
<style__BaseCardWrapper>
  <style__CardWrapperTitle>
    test
  </style__CardWrapperTitle>
  <style__CardWrapperContent
    onScroll={[MockFunction]}
  >
    test
  </style__CardWrapperContent>
</style__BaseCardWrapper>
`;
